docs: clean up world api.md a bit (#1958)
ArchipelagoMW/Archipelago: docs/world api.md

Special items with ID `None` can mark events (read below).

Other classifications include
* `filler`: a regular item or trash item
* `useful`: generally quite useful, but not required for anything logical
* `trap`: negative impact on the player

> * `skip_balancing`: add to `progression` to skip balancing; e.g. currency or tokens

Reviewer: ```suggestion
* `skip_balancing`: adds a `progression` item that skips progression balancing; e.g. currency or tokens
```
I feel like the wording here is confusing

Reviewer: Oh I misunderstood what skip_balancing is then, but it should have `progression_skip_balancing` as well as `skip_balancing`.

Reviewer 2: `progression_skip_balancing` technically isn't a flag; it's a combination of the two flags that's used as shorthand. It still exists and is kind of "expected" use, so probably a good idea to also list it here.
